app/src/views/home/memories_screen.tsx

```diff
@@ -0,0 +1,284 @@
+
+  const toggleThreshold = 100; // Distance to toggle the groups visibility
+
+  const handleScrollEnd = (e: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const offsetY = e.nativeEvent.contentOffset.y;
+    const shouldCollapse = offsetY > toggleThreshold;
+
+    if (shouldCollapse !== isCollapsed) {
+      setIsCollapsed(shouldCollapse);
+      Animated.timing(scrollY, {
+        toValue: shouldCollapse ? 1 : 0,
+        duration: 220, // Reduced animation duration for snappier transitions
+        useNativeDriver: false,
+      }).start();
+    }
+  };
+
+  // Handle restricted navigation for guest users
+  const handleRestrictedAccess = (screenName: string) => {
+    if (userIsGuest) {
+      setShowGuestPopup(screenName);
+    } else {
+      navigation.navigate(screenName);
+    }
+  };
+
+  return (
+    <ThemedView style={styles.bigContainer} testID="memories-screen">
+      <TopBar
+        title="Strive"
+        leftIcon="people-outline"
+        leftAction={() => handleRestrictedAccess("Friends")}
+        rightIcon={
+          userIsGuest || !user.image_id ? "person-circle-outline" : icon
+        }
+        rightAction={() => handleRestrictedAccess("Profile")}
+        testID="top-bar"
+      />
+
+      {/* Challenges */}
+      <Animated.FlatList
+        testID="scroll-view"
+        data={challenges}
+        onScrollEndDrag={handleScrollEnd}
+        keyExtractor={(item, index) => `challenge-${index}`}
+        renderItem={({ item, index }) => (
+          <Challenge
+            navigation={navigation}
+            challengeDB={item}
+            key={index}
+            testID={`challenge-id-${index}`}
+            currentUser={user}
+            index={index}
+          />
+        )}
+        ListHeaderComponent={
+          <ThemedView testID="user-header" style={styles.userHeader}>
+            <ThemedText style={{ fontSize: 20, fontWeight: "bold" }}>
+              {user.name}
+            </ThemedText>
+            <ThemedText style={{ fontSize: 15, fontWeight: "bold" }}>
+              {user.email}
+            </ThemedText>
+          </ThemedView>
+        }
+        ListFooterComponent={
+          userIsGuest && (
+            <View style={styles.guestFooter}>
+              <ThemedText style={styles.guestFooterText}>
+                Create an account to see more challenges!
+              </ThemedText>
+              <TouchableOpacity
+                style={styles.signUpButton}
+                onPress={() => navigation.navigate("SignUp")}
+              >
+                <ThemedText style={styles.signUpButtonText}>Sign Up</ThemedText>
+              </TouchableOpacity>
+            </View>
+          )
+        }
+        ListEmptyComponent={
+          <ThemedText style={styles.noChallengesText}>
+            No challenges to display
+          </ThemedText>
+        }
+        contentContainerStyle={styles.contentContainer}
+      />
+
+      {/* Guest User Pop-Up */}
+      {showGuestPopup && (
```

Comment:
You could restrain the guest access in the home screen to no have to manage this here.